src/walk.js

```diff
@@ -6,8 +6,8 @@ export default make({
     c(node.argument, st, "Expression");
   },
   ObjectPattern(node, st, c) {
-    const list = node.properties;
-    for (const prop of list) {
+    for (const prop of node.properties) {
+      if (prop.computed) c(prop.key, st, "Pattern");
       c(prop.value || prop.argument, st, "Pattern");
     }
```

Comment:
I think we can / should remove ObjectPattern here – acorn-walk has its own implementation that we're overriding and doesn't have the bug this is fixing.

Reply:
Good call! I’ve removed the now unnecessary overrides of ObjectPattern and SpreadPattern (the latter of which is now called SpreadElement).